picard/ui/searchdialog/album.py

```diff
@@ -62,6 +62,7 @@ class CoverWidget(QtWidgets.QWidget):
     def __init__(self, parent, width=100, height=100):
         super().__init__(parent)
         self.layout = QtWidgets.QVBoxLayout(self)
+        self.layout.destroyed.connect(self.invalidate)
```

Comment:
I think this could also connect to `self.destroyed`, as it is the widget itself that primarily is the one getting removed. Would have the same effect, though.

Reply:
Yup, done.